website/source/docs/discovery/consul.html.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: "docs"
+page_title: "Discovery with Consul"
+sidebar_current: "docs-discovery-consul"
+description: >
+  Nomad integrates with Consul to provide seamless service discovery for
+  running jobs and tasks.
+---
+
+# Consul Discovery Integration
+
+[Consul](https://consul.io) is a tool for service discovery produced by
+HashiCorp. It is distributed, highly available, and fault tolerant. Consul
+provides a very convenient DNS interface which can be used to query for
+registered nodes and services. Nomad can leverage Consul's service discovery
+abilities using the built-in Consul discovery provider.
+
+In addition to service discovery, Consul also provides rich health checking
+abilities which directly tie into discovery results. A gossip-based failure
+detector is used to provide near real-time updates of node status and
+failures, and route away from them if possible. This means that by simply
+registering a Nomad task with Consul, you are automatically able to query and
+load balance requests using the DNS interface, and avoid routing to unhealthy
+instances.
+
+## Configuration
+
+This discovery provider allows specifying connectivity settings using Nomad's
+[client options](/docs/agent/config.html#options). The following settins are
+configurable:
+
+* `discovery.consul.enable`: If set to `true`, enables the Consul discovery
+  back-end. Defaults to false.
+* `discovery.consul.address`: Specifies the address of the Consul agent to
+  register with. Defaults to `127.0.0.1:8500`.
+* `discovery.consul.scheme`: Specifies the address scheme to use while
```

Comment:
Is there a reason to separate `address` and `scheme` here? This seems inconsistent with other places where address is used with scheme and address together such as [-address](https://github.com/hashicorp/nomad/blob/fbd7840030b3908c17176183c410d06dbbdfa736/command/meta.go#L87) at the command-line.